backend-deno/samples/latest-value-cache/mod.ts

```diff
@@ -0,0 +1,65 @@
+import { startService } from "https://deno.land/x/telestion/mod.ts";
+import { z } from "https://deno.land/x/zod@v3.21.4/mod.ts";
+
+const { messageBus, config, serviceName } = await startService();
+
+const { DATA_SUBJECT, REQUEST_SUBJECT } = z.object({
+  DATA_SUBJECT: z.string().endsWith(">"),
+  REQUEST_SUBJECT: z.string().endsWith(">"),
+}).parse(config);
+
+console.log("Latest value cache started with config", {
+  DATA_SUBJECT,
+  REQUEST_SUBJECT,
+});
+
+const dataStore = new Map<string, Uint8Array>();
+
+console.log("Listening for new data on", DATA_SUBJECT);
+
+// listen to the data subject
+const data = messageBus.subscribe(DATA_SUBJECT);
+(async () => {
+  for await (const msg of data) {
+    // data key is the ">" in the subject
+    const dataKey = msg.subject.substring(DATA_SUBJECT.length - 1);
+    // store the data
+    dataStore.set(dataKey, msg.data);
+
+    console.log(
+      "Latest value cache received data",
+      msg.data,
+      "for key",
+      dataKey,
+    );
+  }
+})();
+
+// listen to the request subject
+const request = messageBus.subscribe(REQUEST_SUBJECT, {
+  queue: serviceName,
+});
+(async () => {
+  for await (const msg of request) {
+    console.log("Latest value cache received request on", msg.subject);
+
+    // request key is the ">" in the subject
+    const requestKey = msg.subject.substring(REQUEST_SUBJECT.length - 1);
+    // get the data from the store
+    const data = dataStore.get(requestKey);
+    // respond with the data or null if there is no data
+    msg.respond(data);
+
+    console.log(
+      "Latest value cache responded with",
+      data,
+      "for key",
+      requestKey,
+    );
+  }
+})();
+
+await Promise.any([data.closed, request.closed]);
+console.log("Latest value cache stopped");
+data.drain();
+request.drain();
```

Comment:
It runs if, for whatever reason, a subscription gets closed unexpectedly (in which case, the `.closed` promise resolves). But yes, I don't know when this would be the case (maybe if a maximum reconnect count is reached in NATS? IDK)

[![trustme](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/WjlTemxb6oA4PgZFmj08/36b73412-826d-4a8a-9cc2-1ee989dd1179/trustme.gif)](https://app.graphite.dev/macros/wuespace)